kratos/containers/sparse_graph.h

```diff
@@ -0,0 +1,394 @@
+    template<class TContainerType>
+    void AddEntries(const TContainerType& rIndices)
+    {
+        for(auto I : rIndices)
+            mGraph[I].insert(rIndices.begin(), rIndices.end());
+    }
+
+    void AddEntries(const SparseGraph& rOtherGraph)
+    {
+        for(const auto& item: rOtherGraph.GetGraph())
+        {
+            AddEntries(item.first, item.second);
+        }
+    }
+
+    void Finalize()
+    {
+
+    }
+
+    const GraphType& GetGraph() const{
+        return mGraph;
+    }
+
+    IndexType ExportCSRArrays(
+        vector<IndexType>& rRowIndices,
+        vector<IndexType>& rColIndices
+    ) //TODO: this function should be imported in the CSR matrix interface, not here
+    {
+        //need to detect the number of rows this way since there may be gaps
+        IndexType nrows=0;
+        for(const auto& item : this->GetGraph())
+        {
+            nrows = std::max(nrows,item.first+1);
+        }
+
+        if(rRowIndices.size() != nrows+1)
+        {
+            rRowIndices.resize(nrows+1, false);
+        }
+        //set it to zero in parallel to allow first touching
+        #pragma omp parallel for
+        for(int i=0; i<static_cast<int>(nrows+1); ++i)
+            rRowIndices[i] = 0;
+
+        //count the entries TODO: do the loop in parallel if possible
+        for(const auto& item : this->GetGraph())
+        {
+            rRowIndices[item.first+1] = item.second.size();
+        }
+        //sum entries
+        for(int i = 1; i<static_cast<int>(rRowIndices.size()); ++i){
+            rRowIndices[i] += rRowIndices[i-1];
+        }
+
+
+        IndexType nnz = rRowIndices[nrows];
+        if(rColIndices.size() != nnz){
+            rColIndices.resize(nnz, false);
+        }
+        //set it to zero in parallel to allow first touching
+        #pragma omp parallel for
+        for(int i=0; i<static_cast<int>(rColIndices.size());++i)
+            rColIndices[i] = 0;
+
+        //count the entries TODO: do the loop in parallel if possible
+        for(const auto& item : this->GetGraph()){
+            IndexType start = rRowIndices[item.first];
+
+            IndexType counter = 0;
+            for(auto index : item.second){
+                rColIndices[start+counter] = index;
+                counter++;
+            }
+        }
+
+        //reorder columns
+        #pragma omp parallel for
+        for(int i=0; i<static_cast<int>(rRowIndices.size()-1);++i){
+            std::sort(rColIndices.begin()+rRowIndices[i], rColIndices.begin()+rRowIndices[i+1]);
+        }
+        return nrows;
+    }
+
+    ///@}
+    ///@name Operations
+    ///@{
+
+
+    ///@}
+    ///@name Access
+    ///@{
+
+
+    ///@}
+    ///@name Inquiry
+    ///@{
+
+
+    ///@}
+    ///@name Input and output
+    ///@{
+
+    /// Turn back information as a string.
+    virtual std::string Info() const
+    {
+        std::stringstream buffer;
+        buffer << "SparseGraph" ;
+        return buffer.str();
```

Comment:
```suggestion
       return "SparseGraph";
```

Unless you want to add some other information like number of non zeros, etc here

Reply:
yes, i would like to do that longer term

Reply:
it is already improved in #7702